### PR TITLE
361 dexcom integration

### PR DESCRIPTION
**Pull request by Orta21**
Ref: 361

### Dependencies

- Upstream: https://github.com/metriport/metriport-internal/pull/512

### Description

Add Dexcom as a provider to the devices api

### Release Plan

- We will need to update the package versions.
- Need to deploy the secrets

- [x] Upstream dependencies are met
- [ ] Merge once approved

### CODE REVIEW

File: api/app/src/mappings/dexcom/biometrics.ts
Comment (leite08) [inline api/app/src/mappings/dexcom/biometrics.ts:32]: I'm assuming `convert-units` doesn't do this one, right?

File: api/app/src/mappings/dexcom/nutrition.ts
Comment (leite08) [inline api/app/src/mappings/dexcom/nutrition.ts:18]: nit: better to declare consts out of the loop

File: api/app/src/mappings/dexcom/nutrition.ts
Comment (leite08) [inline api/app/src/mappings/dexcom/nutrition.ts:25]: NABD, but we could make this a bit simpler with a `.flatMap()`, no?

File: api/app/src/providers/dexcom.ts
Comment (leite08) [inline api/app/src/providers/dexcom.ts:67]: It would be nice to have these as named constants so we know what these numbers mean when we read the code.

Also, if we're multiplying both by `1000`, we can avoid that multiplication altogether, right?

File: api/app/src/providers/dexcom.ts
Comment (leite08) [inline api/app/src/providers/dexcom.ts:108]: Send to Sentry: `capture.error(...)`

File: api/app/src/providers/dexcom.ts
Comment (leite08) [inline api/app/src/providers/dexcom.ts:87]: IIRC we don't need to send `Date.now()` to `dayjs()`, it already initializes with the current timestamp.

File: api/app/src/providers/dexcom.ts
Comment (leite08) [inline api/app/src/providers/dexcom.ts:136]: ditto dayjs constructor

File: api/app/src/providers/dexcom.ts
Comment (leite08) [inline api/app/src/providers/dexcom.ts:150]: We have a constant for that date format, `ISO_DATE`

File: api/app/src/providers/dexcom.ts
Comment (leite08) [inline api/app/src/providers/dexcom.ts:167]: ditto `ISO_DATE`

File: api/app/src/mappings/dexcom/biometrics.ts
Comment (leite08) [inline api/app/src/mappings/dexcom/biometrics.ts:18]: NABD, but the type of accumulator its inferred from the initialization, which we can use `new Array<Sample>()`. Just mentioning this bc I think its better to have types based on initialization than the other way around - you can see if you remove `: Sample[]` from the `acc` param and hover it, its `never`, which means we're setting the type of a param that receives a value/obj without type, so the compiler can't help us if they mismatch.

**Review by Goncharo — APPROVED** [review]
Looks great - just a reminder to deploy/set the secrets stack and update the GH Action infra config secret before merging

**Comment by leite08** [comment]
Forgot to mention:
- there are conflicts
- we're missing details of the release plan: indication there will be infra updates and what needs to be done upon release; it helps bc:
  - you don't need to be figuring our stuff when deploying, just need to follow the plan;
  - reviewers can help validate the release plan
  - in case someone has to deploy for you, the plan is laid out and reviewed

File: api/app/src/mappings/dexcom/biometrics.ts
Comment (Orta21) [reply api/app/src/mappings/dexcom/biometrics.ts:32]: Yea it doesnt have this conversion

File: api/app/src/providers/dexcom.ts
Comment (leite08) [inline api/app/src/providers/dexcom.ts:117]: Not sure if this is the case, but if we're calling `.getCxIdXXX()` on the route level, we don't need to set the `cxId` on Sentry's context as its already done there (as `user.id`).

If we're not and want to do it here, can we set as `user.id` like `.getCxId()` does, please?

File: api/app/src/providers/withings.ts
Comment (leite08) [inline api/app/src/providers/withings.ts:107]: We have this in two places, maybe worth exposing as a function and calling it from those places?

**Review by leite08 — APPROVED** [review]
LGTM, approving with the assumption we're good with adjusting the `cxId` on capture/Sentry. The other comments are not a huge deal, fine to push without those to move fast if needed.

**Pull request merged**
